split arrow conversion and sync

parmesant opened the pull request:
<!-- Thanks for trying to help us make Parseable be the best it can be! Please fill out as much of the following information as is possible (where relevant, and remove it when irrelevant) to help make the intention and scope of this PR clear in order to ease review. -->

This PR separates out the arrow conversion job from the sync job as we require more visibility into both of them. Arrow to Parquet conversion takes place at the 5th second of every minute and sync/upload happens every 30 seconds. Nothing else has been changed in terms of functionality.

<!-- Replace XXXX with the id of the issue fixed in this PR. Remove this section if there is no corresponding issue. Don't reference the issue in the title of this pull-request. -->

### Description

<!-- Describe the goal of this PR -->

<!-- Describe the possible solutions and chosen one with the rationale. -->

<!-- Describe key changes made in the patch. -->

<hr>

This PR has:
- [ ] been tested to ensure log ingestion and log query works.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added documentation for new or modified features or behaviors.

parmesant committed:
split arrow conversion and sync

parmesant committed:
refactor: linting

parmesant committed:
handled return condition

Returning None based on length of vec
for proper message printing

nikhilsinhaparseable commented on src/storage/object_storage.rs:
```diff
@@ -572,13 +573,17 @@ pub trait ObjectStorage: Debug + Send + Sync + 'static {
         let streams = STREAM_INFO.list_streams();
 
+        // start the sync loop for a stream
+        // parallelize this
```
please implement the code for parallelization for each stream in both sync and conversion flow

nikhilsinhaparseable commented on src/storage/object_storage.rs:
```diff
@@ -587,6 +592,9 @@ pub trait ObjectStorage: Debug + Send + Sync + 'static {
+
+            // read arrow files on disk
+            // convert them to parquet
             let schema = convert_disk_files_to_parquet(
```
why are we still converting arrow to parquet in sync flow

parmesant committed:
add warnings

Upload and conversion tasks are handled by a monitor task which checks whether
these tasks have crossed a threshold or not.
In case either of these tasks crosses a pre-defined threshold, a warning will be printed.

parmesant committed:
parallelize conversion, remove old sync fn

parmesant committed:
modified warn statement

parmesant committed:
refactor: clippy

parmesant committed:
Merge branch 'main' into sync-warning

nikhilsinhaparseable committed:
Merge branch 'main' into sync-warning

Commit:
perf: utilise lesser CPU cycles by actually utilising tokio

Commit:
don't schedule with clokwerk

Commit:
ehance warning message

Commit:
don't include `start_time` in warning

parmesant committed:
Merge pull request #4 from de-sh/sync-side

perf: consume less CPU cycles, by understanding tokio

nitisht committed:
Merge branch 'main' into sync-warning

nitisht committed:
Merge branch 'main' into sync-warning

nitisht committed:
Merge branch 'main' into sync-warning
